### DISCUSSION TITLE
Is there a Serenity-recommended way to handle 404 for a wrong URL sent to the application?

### DISCUSSION THREAD

**Discussion by LSNicholls** [serenity-is/Serenity#6686] [Help]
I am thinking that a wrong route never makes it through to the framework so /Home/Error never gets the info.*

So, should I handle it on the IIS level (just map an error page, old-school), or is there something I can do in program/startup that will allow me to handle it within the application?  

*In other words, try this on the demo:

https://demo.serenity.is/x

gives you the following in Chrome;  I'd like to substitute different behavior:

![image](https://user-images.githubusercontent.com/88156108/218342606-37d10437-e0dc-460a-b985-ec0ffb4ff9f4.png)

Thanks for your thoughts!  Curious about whether other people just leave this as-is or do the 404 map thing in IIS, or what.

**Comment by VictorTomaili** [comment]
You can follow the Microsoft documentation to add error pages to your app
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-6.0

**Comment by LSNicholls** [comment]
Thank you @VictorTomaili , but I was not looking for pointers to Microsoft documentation; in fact I had that page open as I wrote :-).  I just didn't know if there was a Serenity-specific or Serenity-recommended way to handle it that I had missed.   Thank you for confirming that there was no reason that standard behavior *wouldn't* work, although I was confident of that already, because your work is good.

But I will not mark your response as an answer. Maybe I shouldn't even have put this question in the "Help" section; I wrote because I thought it was an interesting discussion to have/share.  If nobody agrees, fine, but let's not shut off the discussion, or dismiss its value. 

The fact that the demo didn't implement anything was a clue.  Other developers' experience would also be important.  I will probably go ahead and implement and contribute my experiences, IAC, because that is what the community is for.

**Reply by VictorTomaili** [reply]
The recommended way is handling it like described in Microsoft documentation. The project is works with kestrel and IIS. When you handle this in IIS, and don't have a plan to use kestrel, it is fine. But other side, when you want to use kestrel, Linux server or docker container, the better way to handle it is in code side.

**Comment by LSNicholls** [comment]
Yep, understood.  Thank you again for confirming there aren't any Serenity-specific pitfalls in doing it that way.

I've used 
`
  app.UseStatusCodePagesWithRedirects("~/StatusCode/{0}");
`
... in case anybody else is interested.  My code is nothing special, but I am happy to share, and to continue the _discussion_, if anybody has had a better experience with one of the other variants.
